### PR TITLE
docs: add documentation concerning URL

### PR DESCRIPTION
**Pull request by julien**

### CODE REVIEW

File: dxp/urls.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+### Use case: appending a parameter in a URL
+
+```
+const myUrl = new URL('http://localhost:8080/?foo=baz');
+
+myUrl.searchParams.append('foo', 'another-baz');
+
+console.log(myUrl.toString());
+
+// Will print: http://localhost:8080/?foo=baz&foo=another-baz
```
Comment (jbalsas) [inline dxp/urls.md]: Maybe we could ellaborate a bit more in which case one would want more than one `foo` param so the use case is clear and not just the technical aspect

File: dxp/urls.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+Make sure you read the documentation referenced above to make sure you fully understand the URL API.
+
+Additionally, a polyfill is also served for Internet Explorer 11, to ensure the `URL` API can be used correcly.
+
+
+### Use case: creating a relative URL
+
+```
+
```
Comment (wincent) [inline dxp/urls.md]: Kill this unnecessary blank line.

File: dxp/urls.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+```
+const myUrl = new URL('http://localhost:8080/?foo=baz&bar=qux');
+
+console.log(myUrl.searchParams.has('foo'));
+console.log(myUrl.searchParams.has('bar'));
+
+// Will print: true (twice)
+
+
```
Comment (wincent) [inline dxp/urls.md]: Kill this blank line. Or perhaps better yet, drop the "will print" comments and do this instead:

```javascript
console.log(myUrl.searchParams.has('foo')); // true
console.log(myUrl.searchParams.has('bar')); // true
console.log(myUrl.searchParams.has('not-there')); // false
```

File: dxp/urls.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+We now recommend using the native [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) API.
+
+Make sure you read the documentation referenced above to make sure you fully understand the URL API.
+
+Additionally, a polyfill is also served for Internet Explorer 11, to ensure the `URL` API can be used correcly.
+
+
+### Use case: creating a relative URL
+
+```
```
Comment (wincent) [inline dxp/urls.md]: Can you please make all of these fenced code blocks start with?

```
```javascript
```

File: dxp/urls.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+// Will print: 'baz'
+
+console.log(myUrl.searchParams.get('bar'));
+
+// Will print: 'qux'
+
+console.log(myUrl.searchParams.get('not-there'));
+
+// Will print: null
+```
```
Comment (wincent) [inline dxp/urls.md]: These also might be better briefer.

```javascript
console.log(myUrl.searchParams.get('foo')); // 'baz'
console.log(myUrl.searchParams.get('bar')); // 'qux'
console.log(myUrl.searchParams.get('not-there')); // null
```

or even?

```javascript
myUrl.searchParams.get('foo'); // 'baz'
myUrl.searchParams.get('bar'); // 'qux'
myUrl.searchParams.get('not-there'); // null
```

"Less is more", and all that. 😂

**Comment by matuzalemsteles** [comment]
Ah my concern in documenting these things is that it seems that we are describing the MDN documentation or duplicating it in some way... Maybe it would be better if it were to compare with `metal-uri` and already link to MDN since it will always be up to date when Something to change.

**Comment by wincent** [comment]
> My concern in documenting these things is that it seems that we are describing the MDN documentation or duplicating it in some way

This is a good point, but:

- The work is already done.
- We don't have to worry about getting "out of sync" with what's on MDN (because there is *no way* the `URL` API is ever going to have breaking changes).
- This document is pretty readable thanks to the headings (compared with the MDN page which just has one big block under "Usage Notes"), so that's useful too.
- We have links to both `metal-uri` (and I wouldn't want to go into more detail about a legacy API, so the link is probably enough) and the MDN pages here.

Which is all a long way of saying I think this change is probably OK.

Let's merge it and anybody reading this thread can feel free to make improvements in another PR.

**Pull request merged**